### PR TITLE
Make conformance tests more strict about the failure list.

### PR DESCRIPTION
**Pull request by haberman**
The failure lists were falling out of date because the
tests would pass even if people forgot to remove failures
from the list.

### CODE REVIEW

**Comment by haberman** [comment]
This PR probably isn't ready to go yet, but I wanted to send it out to get travis testing for it.

**Comment by haberman** [comment]
Tests are passing finally! Review to @xiaofeng and @anandolee (for the `json_format.py` part).

**Comment by xfxyjwf** [comment]
LGTM

I haven't looked at the python changes.

File: python/google/protobuf/json_format.py [hunk 1]
```diff
@@ -555,6 +559,10 @@ def _ConvertScalarFieldValue(value, field, require_str=False):
     if field.type == descriptor.FieldDescriptor.TYPE_BYTES:
       return base64.b64decode(value)
     else:
+      # Checking for unpaired surrogates appears to be unreliable,
+      # depending on the specific Python version, so we check manually.
+      if _UNPAIRED_SURROGATE_PATTERN.search(value):
```
Comment (anandolee) [inline python/google/protobuf/json_format.py]: Add test in json_format_test.py too?

Comment (haberman) [inline python/google/protobuf/json_format.py]: Done, PTAL.

**Comment by anandolee** [comment]
LGTM

**Pull request merged**